datafusion/physical-plan/src/joins/sort_merge_join.rs

```diff
@@ -1254,6 +1261,20 @@ impl SMJStream {
 
                         // For full join, we also need to output the null joined rows from the buffered side
                         if matches!(self.join_type, JoinType::Full) {
+                            // Handle not mask for buffered side further.
+                            // For buffered side, we want to output the rows that are not null joined with
+                            // the streamed side. i.e. the rows that are not null in the `buffered_indices`.
+                            let not_mask = if buffered_indices.null_count() > 0 {
```

Comment:
You might be able to avoid the unwrap by using

```rust 
                            let not_mask = if let Some(nulls) = if buffered_indices.nulls() { 
                                let mask = not_mask.values() & nulls.inner();
                                BooleanArray::new(mask, None)
                            } else {
                                not_mask
                            };
```